modules/gdnative/godot/godot_dictionary.cpp

```diff
@@ -28,109 +28,100 @@
+	Dictionary *self = (Dictionary *)p_self;
+	const Variant *key = (const Variant *)p_key;
+	self->erase(*key);
 }
 
-godot_array GDAPI godot_dictionary_keys(const godot_dictionary *p_dict) {
-	godot_array a;
-	godot_array_new(&a);
-	const Dictionary *dict = (const Dictionary *)p_dict;
-	Array *array = (Array *)&a;
-	*array = dict->keys();
-	return a;
+godot_int GDAPI godot_dictionary_hash(const godot_dictionary *p_self) {
+	const Dictionary *self = (const Dictionary *)p_self;
+	return self->hash();
 }
 
-godot_int GDAPI godot_dictionary_parse_json(godot_dictionary *p_dict, const godot_string *p_json) {
-	Dictionary *dict = (Dictionary *)p_dict;
-	const String *json = (const String *)p_json;
-	Variant ret;
-	int err_line;
-	String err_str;
-	int err = (int)JSON::parse(*json, ret, err_str, err_line);
-	*dict = ret;
-	return err;
+godot_array GDAPI godot_dictionary_keys(const godot_dictionary *p_self) {
+	godot_array dest;
+	const Dictionary *self = (const Dictionary *)p_self;
+	memnew_placement(&dest, Array(self->keys()));
+	return dest;
 }
 
-godot_variant GDAPI *godot_dictionary_operator_index(godot_dictionary *p_dict, const godot_variant *p_key) {
-	Dictionary *dict = (Dictionary *)p_dict;
-	Variant *key = (Variant *)p_key;
-	return (godot_variant *)&dict->operator[](*key);
+godot_array GDAPI godot_dictionary_values(const godot_dictionary *p_self) {
+	godot_array dest;
+	const Dictionary *self = (const Dictionary *)p_self;
+	memnew_placement(&dest, Array(self->values()));
+	return dest;
 }
 
-godot_int GDAPI godot_dictionary_size(const godot_dictionary *p_dict) {
+godot_variant GDAPI godot_dictionary_operator_index(godot_dictionary *p_dict, const godot_variant *p_key) {
+	godot_variant raw_dest;
+	Variant *dest = (Variant *)&raw_dest;
 	const Dictionary *dict = (const Dictionary *)p_dict;
-	return dict->size();
+	const Variant *key = (const Variant *)p_key;
+	*dest = dict->operator[](*key);
+	return raw_dest;
 }
 
-godot_string GDAPI godot_dictionary_to_json(const godot_dictionary *p_dict) {
-	const Dictionary *dict = (const Dictionary *)p_dict;
-	godot_string str;
-	godot_string_new(&str);
-	String *s = (String *)&str;
-	*s = JSON::print(Variant(*dict));
-	return str;
+godot_bool GDAPI godot_dictionary_operator_equal(const godot_dictionary *p_self, const godot_dictionary *p_b) {
+	const Dictionary *self = (const Dictionary *)p_self;
+	const Dictionary *b = (const Dictionary *)p_b;
+	return *self == *b;
 }
 
-godot_array GDAPI godot_dictionary_values(const godot_dictionary *p_dict) {
-	godot_array a;
-	godot_array_new(&a);
+godot_string GDAPI godot_dictionary_to_json(const godot_dictionary *p_dict) {
+	godot_string raw_dest;
+	String *dest = (String *)&raw_dest;
 	const Dictionary *dict = (const Dictionary *)p_dict;
-	Array *array = (Array *)&a;
-	*array = dict->values();
-	return a;
-}
-
-void GDAPI godot_dictionary_destroy(godot_dictionary *p_dict) {
-	((Dictionary *)p_dict)->~Dictionary();
+	*dest = JSON::print(Variant(*dict));
```

Comment:
This is buggy code. The `godot_string_new()` was there for a reason. String does some heap allocations and also has an operator overload for `=`, so the object needs to be valid. This is undefined behaviour.